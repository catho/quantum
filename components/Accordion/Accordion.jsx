import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import slugify from 'slugify';

import {
  colors,
  spacing as defaultSpacing,
  baseFontSize as defaultBaseFontSize,
} from '../shared/theme';
import { shadow } from '../shared';
import Icon from '../Icon';

const Wrapper = styled.ul`
  ${({
    theme: {
      spacing: { medium },
      baseFontSize,
      colors: {
        neutral: { 900: color },
      },
    },
  }) => `
    color: ${color};
    font-size: ${baseFontSize}px;
    padding: 0 ${medium}px;
  `}
  background-color: inherit;
  list-style-type: none;
`;

const AccordionTitle = styled.span`
  ${({
    theme: {
      baseFontSize,
      spacing: { medium },
      colors: { neutral },
    },
  }) => `
    color: ${neutral[700]};
    font-size: ${baseFontSize}px;
    margin-left: ${medium}px;
  `}

  font-weight: bold;
  line-height: inherit;
`;

const AccordionHeader = styled.button.attrs({ type: 'button' })`
  ${({ spacing: { large, medium } }) => `
    height: ${large}px;
    line-height: ${large}px;
    padding: ${medium}px 0;
  `}
  background: inherit;
  border-radius: 4px;
  border: none;
  box-sizing: content-box;
  cursor: pointer;
  display: inline-flex;
  justify-content: space-between;
  width: 100%;

  ${shadow(1)}

  ${({ as }) => (as === 'button' ? `cursor: pointer;` : null)}
`;

const AccordionContent = styled.div`
  ${shadow(1)}

  ${({
    theme: {
      spacing: { xsmall, medium },
      baseFontSize,
      colors: { primary, secondary },
    },
    opened,
  }) => `
    border-left: 4px solid ${primary[700]};
    border-radius: 4px;
    display: ${opened ? 'block' : 'none'}; 
    font-size: ${baseFontSize - 2}px;
    height: ${opened ? 'auto' : '0'};
    margin: ${medium}px 0;
    overflow: hidden;
    padding: ${medium}px;
    

    a {
      color: ${primary[700]};
      font-size: ${baseFontSize - 2}px;
      text-decoration: underline;

      &:hover {
        color: ${secondary[700]};
      }
    }

    ul {
      list-style-type: none;
      padding: 0;

      & > li {
        padding: ${xsmall}px 0;
      }
    }

    p {
      margin: 0;
    }
  `}
`;

const StyledIcon = styled(Icon)`
  ${({
    theme: {
      baseFontSize,
      spacing: { large },
      colors: { primary },
    },
  }) => `
    font-size: ${baseFontSize * 1.5}px;
    margin-right: ${large}px;
    color: ${primary[700]};
  `}

  line-height: inherit;
`;

AccordionHeader.displayName = 'AccordionHeader';
AccordionContent.displayName = 'AccordionContent';

class Accordion extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      items: props.items,
    };
  }

  setAccordionItems(items) {
    this.setState({ items });
  }

  toggleItem = indexToChange => {
    const { items } = this.state;
    const { keepOnlyOneOpen } = this.props;
    let newItems = items;

    if (keepOnlyOneOpen) {
      newItems = this.closeAllContents();
    }

    newItems = this.toggleItemVisibility(newItems, indexToChange);

    this.setAccordionItems(newItems);
  };

  handleCustomItem = items =>
    items.map((item, index) => {
      const customItem = item;
      customItem.onItemClick = i => {
        if (typeof item.onClick === 'function') {
          item.onClick();
        }
        this.toggleItem(i);
      };
      customItem.opened = !!item.opened;

      return this.renderItem(customItem, index);
    });

  closeAllContents() {
    const { items } = this.state;
    return items.map(item => ({ ...item, opened: false }));
  }

  toggleItemVisibility(newItems, indexToChange) {
    const { items } = this.state;
    const itemsToReturn = newItems;
    itemsToReturn[indexToChange].opened = !items[indexToChange].opened;

    return itemsToReturn;
  }

  renderItem(item, itemIndex) {
    const { title, content, opened, onItemClick } = item;
    const { theme } = this.props;
    const {
      colors: {
        neutral: { 900: color },
      },
      spacing,
    } = theme;
    const itemId = `${slugify(title)}-${itemIndex}`;
    const contentId = `${itemId}-content`;
    const headerId = `${itemId}-header`;

    return (
      <li className="accordion-item" key={title} opened={opened} color={color}>
        <AccordionHeader
          spacing={spacing}
          onClick={() => onItemClick(itemIndex)}
          aria-controls={contentId}
          id={headerId}
          theme={theme}
        >
          <AccordionTitle theme={theme}>{title}</AccordionTitle>
          <StyledIcon
            theme={theme}
            name={opened ? 'expand_less' : 'expand_more'}
          />
        </AccordionHeader>
        {content && (
          <AccordionContent
            id={contentId}
            aria-labelledby={headerId}
            opened={opened}
            aria-hidden={!opened}
            theme={theme}
          >
            {content}
          </AccordionContent>
        )}
      </li>
    );
  }

  render() {
    const { theme } = this.props;
    const { items } = this.state;

    return <Wrapper theme={theme}>{this.handleCustomItem(items)}</Wrapper>;
  }
}

Accordion.defaultProps = {
  theme: {
    colors,
    spacing: defaultSpacing,
    baseFontSize: defaultBaseFontSize,
  },
  keepOnlyOneOpen: false,
};

Accordion.propTypes = {
  /** Theme natively passed by Quantum theme. Don't change this prop */
  theme: PropTypes.shape({
    colors: PropTypes.object,
    spacing: PropTypes.object,
    baseFontSize: PropTypes.number,
  }),
  /** Set true to keep only one item open and close the remainings on click */
  keepOnlyOneOpen: PropTypes.bool,
  /**
   * this prop receives an array of objects to render the content of component.
   *
   * Ps 1: The content item should receive HTML elements
   *
   * Ps 2: onClick item receives a callback function its will trigger when the respective item of Accordion is clicked */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      content: PropTypes.node,
      opened: PropTypes.bool,
      onClick: PropTypes.func,
    }),
  ).isRequired,
};

export default Accordion;
