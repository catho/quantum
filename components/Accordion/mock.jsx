/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';

const institucionalItem = {
  title: 'Institucional',
  content: (
    <ul>
      <li>
        <a href="#">Blog</a>
      </li>
      <li>
        <a href="#">Contatos</a>
      </li>
      <li>
        <a href="#">Certificação</a>
      </li>
      <li>
        <a href="#">Imprensa</a>
      </li>
      <li>
        <a href="#">Mapa do site</a>
      </li>
    </ul>
  ),
};

const candidateItemOpened = {
  title: 'Para candidato',
  content: (
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      <br />
      Ut orci metus, auctor et viverra nec, consectetur vel diam.
      <br />
      Pellentesque non dui id ex fringilla posuere. Donec at hendrerit leo.
      <br />
      Aenean sit amet velit est.Aliquam erat volutpat.
      <br />
      In eu luctus ipsum. Etiam nibh neque, posuere sit amet euismod sit amet,
      <br />
      fringilla non nisi. Proin in pharetra tellus.
    </p>
  ),
  opened: true,
};

const companiesItem = {
  title: 'Para Empresa',
  content: (
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      <br />
      Ut orci metus, auctor et viverra nec, consectetur vel diam.
      <br />
      Pellentesque non dui id ex fringilla posuere. Donec at hendrerit leo.
      <br />
      Aenean sit amet velit est.Aliquam erat volutpat.
      <br />
      In eu luctus ipsum. Etiam nibh neque, posuere sit amet euismod sit amet,
      <br />
      fringilla non nisi. Proin in pharetra tellus.
    </p>
  ),
};

const educationItem = {
  title: 'Educação',
  content: (
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      <br />
      Ut orci metus, auctor et viverra nec, consectetur vel diam.
      <br />
      Pellentesque non dui id ex fringilla posuere. Donec at hendrerit leo.
      <br />
      Aenean sit amet velit est.Aliquam erat volutpat.
      <br />
      In eu luctus ipsum. Etiam nibh neque, posuere sit amet euismod sit amet,
      <br />
      fringilla non nisi. Proin in pharetra tellus.
    </p>
  ),
  onClick: () => {
    console.log('Educação event triggered');
  },
};

const mock = {
  items: [
    { ...institucionalItem },
    { ...candidateItemOpened },
    { ...companiesItem },
    { ...educationItem },
  ],
};

const mockWithFirstItemOpened = {
  items: [
    { ...candidateItemOpened },
    { ...institucionalItem },
    { ...companiesItem },
    { ...educationItem },
  ],
};

const mockWithLastItemOpened = {
  items: [
    { ...institucionalItem },
    { ...companiesItem },
    { ...educationItem },
    { ...candidateItemOpened },
  ],
};

const mockWithAllItemsClosed = {
  items: [{ ...institucionalItem }, { ...companiesItem }, { ...educationItem }],
};

const mockCode = `
  items: [
    {
      title: 'Institucional',
      content: (
        <ul>
          <li>
            <a href="#">Blog</a>
          </li>
          <li>
            <a href="#">Contatos</a>
          </li>
          ...
        </ul>
      ),
    },
    {
      title: 'Para candidato',
      content: (
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          <br />
          Ut orci metus, auctor et viverra nec, consectetur vel diam.
          ...
        </p>
      ),
      opened: true,
    },
    {
      title: 'Para Empresa',
      content: (
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          <br />
          Ut orci metus, auctor et viverra nec, consectetur vel diam.
          <br />
          Pellentesque non dui id ex fringilla posuere. Donec at hendrerit leo.
          ...
        </p>
      ),
    },
    {
      title: 'Educação',
      content: (
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          <br />
          Ut orci metus, auctor et viverra nec, consectetur vel diam.
          ...
        </p>
      ),
    },
  ];
`;

export {
  mock as default,
  mockCode,
  mockWithFirstItemOpened,
  mockWithLastItemOpened,
  mockWithAllItemsClosed,
};
