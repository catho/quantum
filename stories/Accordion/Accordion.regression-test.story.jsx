import React from 'react';
import { storiesOf } from '@storybook/react';
import mockedProps, {
  mockWithFirstItemOpened,
  mockWithLastItemOpened,
  mockWithAllItemsClosed,
} from '../../components/Accordion/mock';

import { Accordion } from '../../components';

storiesOf('Accordion', module)
  .add('Accordion default', () => <Accordion {...mockedProps} />)
  .add('Accordion with first item opened', () => (
    <Accordion {...mockWithFirstItemOpened} />
  ))
  .add('Accordion with last item opened', () => (
    <Accordion {...mockWithLastItemOpened} />
  ))
  .add('Accordion with all items closed', () => (
    <Accordion {...mockWithAllItemsClosed} />
  ));
