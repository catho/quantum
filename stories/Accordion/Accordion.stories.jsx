import mockedProps from '../../test-utils/mocks/Accordion/mock';
import { Accordion } from '../../components';

export default {
  title: 'Accordion',
  component: Accordion,
};

const Template = (args) => <Accordion {...mockedProps} {...args} />;

export const Default = Template.bind({});

export const KeepOnlyOneOpen = Template.bind({});
KeepOnlyOneOpen.args = {
  keepOnlyOneOpen: true,
};
