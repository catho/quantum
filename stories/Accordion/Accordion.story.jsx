import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  Heading,
  Tab,
  SimpleHighlight,
  Title,
  StoryContainer,
  AutoPropsApi,
} from '@catho/quantum-storybook-ui';
import mockedProps, { mockCode } from '../../components/Accordion/mock';

import { Container, Row, Col, TabbedView, Accordion } from '../../components';
import accordionExamples from './examples';

const description =
  "An accordion is a vertical stack of interactive headings used to toggle the display of further information; each item can be 'collapsed', with just a short label visible, or 'expanded' to show the complete content.";

storiesOf('Accordion', module).add('Accordion', () => (
  <>
    <Heading name="Accordion">{description}</Heading>
    <TabbedView>
      <Tab title="Usage">
        <StoryContainer>
          <Title as="h2">Importing Accordion</Title>
          <SimpleHighlight>
            {`import { Accordion } from '@catho/quantum';`}
          </SimpleHighlight>
          <br />
          <Accordion {...mockedProps} keepOnlyOneOpen />
        </StoryContainer>
      </Tab>
      <Tab title="API">
        <StoryContainer>
          <AutoPropsApi component={Accordion} />
        </StoryContainer>
      </Tab>
      <Tab title="Examples">
        <Container no-gutters>
          <Row key="code-example">
            <Col xsmall={2} small={4} medium={6}>
              <SimpleHighlight>{mockCode}</SimpleHighlight>
            </Col>
            <Col xsmall={2} small={4} medium={6}>
              <h3>Example of Object passed in Items</h3>
              <p> This is a sample of object items passed by Items prop</p>
            </Col>
          </Row>
        </Container>
        <Container>
          <h4>
            passing this object to the items prop, the component will render
            this way:
          </h4>
        </Container>
        <Container no-gutters>
          {accordionExamples.map(example => (
            <Container no-gutters key={example.code}>
              <h4>{example.title}</h4>
              <Row key={example.code}>
                <Col xsmall={2} small={4} medium={6}>
                  <SimpleHighlight>{example.code}</SimpleHighlight>
                </Col>
                <Col xsmall={2} small={4} medium={6}>
                  {example.component}
                </Col>
              </Row>
            </Container>
          ))}
        </Container>
      </Tab>
    </TabbedView>
  </>
));
