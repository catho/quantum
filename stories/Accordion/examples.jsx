import React from 'react';
import mockedProps from '../../components/Accordion/mock';

import { Accordion } from '../../components';

const defaultExample = {
  component: <Accordion {...mockedProps} />,
  code: `
<Accordion items={items} />
  `,
};
const keepOnlyOneOpenExample = {
  component: <Accordion {...mockedProps} keepOnlyOneOpen />,
  code: `
<Accordion items={items} keepOnlyOneOpen />
  `,
};

export default [defaultExample, keepOnlyOneOpenExample];
