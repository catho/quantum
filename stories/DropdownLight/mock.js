export const itemsObjectMock = [
  { value: 'Lemon', label: 'Lemon' },
  { value: 'Lime', label: 'Lime' },
  { value: 'Banana', label: 'Banana' },
  { value: 'Açaí', label: 'Açaí' },
  { value: 'Strawberry', label: 'Strawberry' },
  { value: 'Orange', label: 'Orange' },
  { value: 'Avocado', label: 'Avocado' },
  { value: 'Acerola', label: 'Acerola' },
];

export const itemsStringMock = [
  'Lemon',
  'Banana',
  'Strawberry',
  'Orange',
  'Avocado',
  'Açaí',
];

export const itemsWithImageMock = [
  {
    label: 'Master Card',
    value: 'Master Card',
    img: 'https://dummyimage.com/24x24',
    alt: 'image description',
  },
  {
    label: 'Hipercard',
    value: 'Hipercard',
    img: 'https://dummyimage.com/24x24',
    alt: 'image description',
  },
  {
    label: 'American Express Card',
    value: 'American Express Card',
    img: 'https://dummyimage.com/24x24',
    alt: 'image description',
  },
  {
    label: 'Visa',
    value: 'Visa',
    img: 'https://dummyimage.com/24x24',
    alt: 'image description',
  },
  {
    label: 'Elo',
    value: 'Elo',
    img: 'https://dummyimage.com/24x24',
    alt: 'image description',
  },
];
