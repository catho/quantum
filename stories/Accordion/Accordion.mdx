import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';

import { Accordion } from '../../components';

import { Header } from '../shared';

import { 
  Default,
  KeepOnlyOneOpen,
} from './Accordion.stories.jsx';
import EvaluationForm from '../../stories/shared/EvaluationForm';


<Meta title="Accordion" component={Accordion}  />

<Header title="Accordion">
  An Accordion is a vertical stack of clickable headings that are used to toggle information that requires interaction to be shown. Each item can be 'collapsed', displaying just a short visible label; or 'expanded', to show the entire content.
</Header>

- [Importing and usage](#importing)
- [API](#api)
- [Examples](#examples)

## <a id="importing"></a>Importing and usage

```js
import Accordion from '@catho/quantum/Accordion';
```

<Canvas>
  <Story of={Default} />
</Canvas>

## <a id="api"></a>API

<ArgsTable of={Accordion} />

## <a id="examples"></a>Examples

### Keep only one open

<Canvas>
  <Story of={KeepOnlyOneOpen} />
</Canvas>

<EvaluationForm componentName="Accordion" /> 
