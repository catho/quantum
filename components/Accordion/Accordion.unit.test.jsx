import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Accordion from './Accordion';
import mock from './mock';

const closeMock = m => {
  const newMock = m;
  newMock.items = m.items.map(item => ({ ...item, opened: false }));
  return newMock;
};

const getAccordionItemByPosition = (index, el) =>
  el.find('AccordionItem').at(index);

describe('Accordion component', () => {
  it('Should match the snapshot of a simple circular loader', () => {
    const simpleAccordion = <Accordion {...closeMock(mock)} />;
    expect(renderer.create(simpleAccordion).toJSON()).toMatchSnapshot();
  });

  it('Should open and close items and change area-hidden property', () => {
    const wrapper = shallow(<Accordion {...closeMock(mock)} />);
    const getFirstItem = () => wrapper.find('AccordionItem').first();
    const getFirstItemContentAriaHidden = () =>
      getFirstItem()
        .find('AccordionContent')
        .prop('aria-hidden');
    expect(getFirstItem().prop('opened')).toEqual(false);
    expect(getFirstItemContentAriaHidden()).toEqual(true);
    getFirstItem()
      .find('AccordionHeader')
      .simulate('click');
    expect(getFirstItem().prop('opened')).toEqual(true);
    expect(getFirstItemContentAriaHidden()).toEqual(false);
  });

  it('Should open more than one item', () => {
    const wrapper = shallow(<Accordion {...closeMock(mock)} />);
    for (let i = 0; i < 2; i += 1) {
      expect(getAccordionItemByPosition(i, wrapper).prop('opened')).toEqual(
        false,
      );
      getAccordionItemByPosition(i, wrapper)
        .find('AccordionHeader')
        .simulate('click');
      expect(getAccordionItemByPosition(i, wrapper).prop('opened')).toEqual(
        true,
      );
    }
  });

  it('Should close another items when `keepOnlyOneOpen` is true ', () => {
    const wrapper = shallow(<Accordion keepOnlyOneOpen {...closeMock(mock)} />);
    for (let i = 0; i < 2; i += 1) {
      expect(getAccordionItemByPosition(i, wrapper).prop('opened')).toEqual(
        false,
      );
      getAccordionItemByPosition(i, wrapper)
        .find('AccordionHeader')
        .simulate('click');
      expect(getAccordionItemByPosition(i, wrapper).prop('opened')).toEqual(
        true,
      );

      if (i > 0) {
        const j = i - 1;
        expect(getAccordionItemByPosition(j, wrapper).prop('opened')).toEqual(
          false,
        );
      }
    }
  });

  it('Should call onClick function', () => {
    const onClickMock = jest.fn();
    const onCloseMock = closeMock(mock);
    onCloseMock.items[0].onClick = onClickMock;
    const wrapper = shallow(<Accordion {...onCloseMock} />);
    const getFirstItem = () => wrapper.find('AccordionItem').first();
    getFirstItem()
      .find('AccordionHeader')
      .simulate('click');
    expect(onClickMock).toBeCalled();
  });
});
