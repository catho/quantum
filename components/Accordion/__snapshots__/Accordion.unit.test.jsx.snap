// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component Should match the snapshot of a simple circular loader 1`] = `
.c0 {
  font-size: 16px;
  padding: 0 16px;
  color: #191919;
  list-style-type: none;
  background-color: inherit;
}

.c1 {
  border-bottom: 1px solid #191919;
}

.c1:last-child {
  border-bottom: 1px solid transparent;
}

.c3 {
  font-size: 16px;
  font-weight: bold;
  line-height: inherit;
}

.c2 {
  line-height: 24px;
  height: 24px;
  padding: 16px 0;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  background: inherit;
  border: none;
  box-sizing: content-box;
}

.c5 {
  height: 0;
  -webkit-transform: scaleY(0);
  -ms-transform: scaleY(0);
  transform: scaleY(0);
  -webkit-transform-origin: top;
  -ms-transform-origin: top;
  transform-origin: top;
  overflow: hidden;
  -webkit-transition: -webkit-transform 0.2s ease;
  -webkit-transition: transform 0.2s ease;
  transition: transform 0.2s ease;
  font-size: 14px;
}

.c5 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 14px;
}

.c5 ul {
  list-style-type: none;
  padding: 0;
}

.c5 ul > li {
  padding: 8px 0;
}

.c5 p {
  margin: 0;
}

.c4 {
  font-size: 24px;
  line-height: inherit;
}

<ul
  className="c0"
>
  <li
    className="c1"
    color="#191919"
  >
    <button
      aria-controls="Institucional-0-content"
      className="c2"
      id="Institucional-0-header"
      onClick={[Function]}
      spacing={
        Object {
          "large": 24,
          "medium": 16,
          "small": 12,
          "xlarge": 32,
          "xsmall": 8,
          "xxlarge": 40,
          "xxsmall": 4,
          "xxxlarge": 48,
          "xxxsmall": 2,
        }
      }
      type="button"
    >
      <span
        className="c3"
      >
        Institucional
      </span>
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root c4"
        style={
          Object {
            "color": "",
          }
        }
        theme={
          Object {
            "baseFontSize": 16,
            "colors": Object {
              "error": Object {
                "100": "#fff5f5",
                "300": "#f88e86",
                "500": "#ec584e",
                "700": "#e91b0c",
                "900": "#ac001a",
              },
              "neutral": Object {
                "0": "#ffffff",
                "100": "#f2f2f2",
                "1000": "#000000",
                "300": "#e0e0e0",
                "500": "#999999",
                "700": "#424242",
                "900": "#191919",
              },
              "primary": Object {
                "100": "#E5EDFC",
                "300": "#89AAE7",
                "500": "#0CC0EA",
                "700": "#1250C4",
                "900": "#002F7B",
              },
              "secondary": Object {
                "100": "#F7B0C8",
                "300": "#F278A1",
                "500": "#E91E63",
                "700": "#DE0059",
                "900": "#9E003F",
              },
              "success": Object {
                "100": "#dcedc8",
                "300": "#c9eda2",
                "500": "#7ed321",
                "700": "#3b610f",
                "900": "#004D40",
              },
              "warning": Object {
                "100": "#ffefd6",
                "300": "#ffda6a",
                "500": "#ffc107",
                "700": "#f09100",
                "900": "#d14900",
              },
            },
            "spacing": Object {
              "large": 24,
              "medium": 16,
              "small": 12,
              "xlarge": 32,
              "xsmall": 8,
              "xxlarge": 40,
              "xxsmall": 4,
              "xxxlarge": 48,
              "xxxsmall": 2,
            },
          }
        }
      >
        expand_more
      </span>
    </button>
    <div
      aria-hidden={true}
      aria-labelledby="Institucional-0-header"
      className="c5"
      id="Institucional-0-content"
    >
      <ul>
        <li>
          <a
            href="#"
          >
            Blog
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Contatos
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Certificação
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Imprensa
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Mapa do site
          </a>
        </li>
      </ul>
    </div>
  </li>
  <li
    className="c1"
    color="#191919"
  >
    <button
      aria-controls="Para-candidato-1-content"
      className="c2"
      id="Para-candidato-1-header"
      onClick={[Function]}
      spacing={
        Object {
          "large": 24,
          "medium": 16,
          "small": 12,
          "xlarge": 32,
          "xsmall": 8,
          "xxlarge": 40,
          "xxsmall": 4,
          "xxxlarge": 48,
          "xxxsmall": 2,
        }
      }
      type="button"
    >
      <span
        className="c3"
      >
        Para candidato
      </span>
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root c4"
        style={
          Object {
            "color": "",
          }
        }
        theme={
          Object {
            "baseFontSize": 16,
            "colors": Object {
              "error": Object {
                "100": "#fff5f5",
                "300": "#f88e86",
                "500": "#ec584e",
                "700": "#e91b0c",
                "900": "#ac001a",
              },
              "neutral": Object {
                "0": "#ffffff",
                "100": "#f2f2f2",
                "1000": "#000000",
                "300": "#e0e0e0",
                "500": "#999999",
                "700": "#424242",
                "900": "#191919",
              },
              "primary": Object {
                "100": "#E5EDFC",
                "300": "#89AAE7",
                "500": "#0CC0EA",
                "700": "#1250C4",
                "900": "#002F7B",
              },
              "secondary": Object {
                "100": "#F7B0C8",
                "300": "#F278A1",
                "500": "#E91E63",
                "700": "#DE0059",
                "900": "#9E003F",
              },
              "success": Object {
                "100": "#dcedc8",
                "300": "#c9eda2",
                "500": "#7ed321",
                "700": "#3b610f",
                "900": "#004D40",
              },
              "warning": Object {
                "100": "#ffefd6",
                "300": "#ffda6a",
                "500": "#ffc107",
                "700": "#f09100",
                "900": "#d14900",
              },
            },
            "spacing": Object {
              "large": 24,
              "medium": 16,
              "small": 12,
              "xlarge": 32,
              "xsmall": 8,
              "xxlarge": 40,
              "xxsmall": 4,
              "xxxlarge": 48,
              "xxxsmall": 2,
            },
          }
        }
      >
        expand_more
      </span>
    </button>
    <div
      aria-hidden={true}
      aria-labelledby="Para-candidato-1-header"
      className="c5"
      id="Para-candidato-1-content"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        <br />
        Ut orci metus, auctor et viverra nec, consectetur vel diam.
        <br />
        Pellentesque non dui id ex fringilla posuere. Donec at hendrerit leo.
        <br />
        Aenean sit amet velit est.Aliquam erat volutpat.
        <br />
        In eu luctus ipsum. Etiam nibh neque, posuere sit amet euismod sit amet,
        <br />
        fringilla non nisi. Proin in pharetra tellus.
      </p>
    </div>
  </li>
  <li
    className="c1"
    color="#191919"
  >
    <button
      aria-controls="Para-Empresa-2-content"
      className="c2"
      id="Para-Empresa-2-header"
      onClick={[Function]}
      spacing={
        Object {
          "large": 24,
          "medium": 16,
          "small": 12,
          "xlarge": 32,
          "xsmall": 8,
          "xxlarge": 40,
          "xxsmall": 4,
          "xxxlarge": 48,
          "xxxsmall": 2,
        }
      }
      type="button"
    >
      <span
        className="c3"
      >
        Para Empresa
      </span>
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root c4"
        style={
          Object {
            "color": "",
          }
        }
        theme={
          Object {
            "baseFontSize": 16,
            "colors": Object {
              "error": Object {
                "100": "#fff5f5",
                "300": "#f88e86",
                "500": "#ec584e",
                "700": "#e91b0c",
                "900": "#ac001a",
              },
              "neutral": Object {
                "0": "#ffffff",
                "100": "#f2f2f2",
                "1000": "#000000",
                "300": "#e0e0e0",
                "500": "#999999",
                "700": "#424242",
                "900": "#191919",
              },
              "primary": Object {
                "100": "#E5EDFC",
                "300": "#89AAE7",
                "500": "#0CC0EA",
                "700": "#1250C4",
                "900": "#002F7B",
              },
              "secondary": Object {
                "100": "#F7B0C8",
                "300": "#F278A1",
                "500": "#E91E63",
                "700": "#DE0059",
                "900": "#9E003F",
              },
              "success": Object {
                "100": "#dcedc8",
                "300": "#c9eda2",
                "500": "#7ed321",
                "700": "#3b610f",
                "900": "#004D40",
              },
              "warning": Object {
                "100": "#ffefd6",
                "300": "#ffda6a",
                "500": "#ffc107",
                "700": "#f09100",
                "900": "#d14900",
              },
            },
            "spacing": Object {
              "large": 24,
              "medium": 16,
              "small": 12,
              "xlarge": 32,
              "xsmall": 8,
              "xxlarge": 40,
              "xxsmall": 4,
              "xxxlarge": 48,
              "xxxsmall": 2,
            },
          }
        }
      >
        expand_more
      </span>
    </button>
    <div
      aria-hidden={true}
      aria-labelledby="Para-Empresa-2-header"
      className="c5"
      id="Para-Empresa-2-content"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        <br />
        Ut orci metus, auctor et viverra nec, consectetur vel diam.
        <br />
        Pellentesque non dui id ex fringilla posuere. Donec at hendrerit leo.
        <br />
        Aenean sit amet velit est.Aliquam erat volutpat.
        <br />
        In eu luctus ipsum. Etiam nibh neque, posuere sit amet euismod sit amet,
        <br />
        fringilla non nisi. Proin in pharetra tellus.
      </p>
    </div>
  </li>
  <li
    className="c1"
    color="#191919"
  >
    <button
      aria-controls="Educacao-3-content"
      className="c2"
      id="Educacao-3-header"
      onClick={[Function]}
      spacing={
        Object {
          "large": 24,
          "medium": 16,
          "small": 12,
          "xlarge": 32,
          "xsmall": 8,
          "xxlarge": 40,
          "xxsmall": 4,
          "xxxlarge": 48,
          "xxxsmall": 2,
        }
      }
      type="button"
    >
      <span
        className="c3"
      >
        Educação
      </span>
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root c4"
        style={
          Object {
            "color": "",
          }
        }
        theme={
          Object {
            "baseFontSize": 16,
            "colors": Object {
              "error": Object {
                "100": "#fff5f5",
                "300": "#f88e86",
                "500": "#ec584e",
                "700": "#e91b0c",
                "900": "#ac001a",
              },
              "neutral": Object {
                "0": "#ffffff",
                "100": "#f2f2f2",
                "1000": "#000000",
                "300": "#e0e0e0",
                "500": "#999999",
                "700": "#424242",
                "900": "#191919",
              },
              "primary": Object {
                "100": "#E5EDFC",
                "300": "#89AAE7",
                "500": "#0CC0EA",
                "700": "#1250C4",
                "900": "#002F7B",
              },
              "secondary": Object {
                "100": "#F7B0C8",
                "300": "#F278A1",
                "500": "#E91E63",
                "700": "#DE0059",
                "900": "#9E003F",
              },
              "success": Object {
                "100": "#dcedc8",
                "300": "#c9eda2",
                "500": "#7ed321",
                "700": "#3b610f",
                "900": "#004D40",
              },
              "warning": Object {
                "100": "#ffefd6",
                "300": "#ffda6a",
                "500": "#ffc107",
                "700": "#f09100",
                "900": "#d14900",
              },
            },
            "spacing": Object {
              "large": 24,
              "medium": 16,
              "small": 12,
              "xlarge": 32,
              "xsmall": 8,
              "xxlarge": 40,
              "xxsmall": 4,
              "xxxlarge": 48,
              "xxxsmall": 2,
            },
          }
        }
      >
        expand_more
      </span>
    </button>
    <div
      aria-hidden={true}
      aria-labelledby="Educacao-3-header"
      className="c5"
      id="Educacao-3-content"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        <br />
        Ut orci metus, auctor et viverra nec, consectetur vel diam.
        <br />
        Pellentesque non dui id ex fringilla posuere. Donec at hendrerit leo.
        <br />
        Aenean sit amet velit est.Aliquam erat volutpat.
        <br />
        In eu luctus ipsum. Etiam nibh neque, posuere sit amet euismod sit amet,
        <br />
        fringilla non nisi. Proin in pharetra tellus.
      </p>
    </div>
  </li>
</ul>
`;
